packages/expo-ads-admob/android/src/main/java/expo/modules/ads/admob/AdMobInterstitialAdModule.java

```diff
@@ -72,124 +76,92 @@ public void setAdUnitID(String adUnitID, final Promise promise) {
-            adRequestBuilder = adRequestBuilder.addTestDevice(testDeviceID);
+    new Handler(Looper.getMainLooper()).post(() -> {
+      if (mInterstitialAd != null) {
+        promise.reject("E_AD_ALREADY_LOADED", "Ad is already loaded.", null);
+        return;
+      }
+
+      mRequestAdPromise = promise;
+      AdRequest adRequest = new AdRequest.Builder()
+          .addNetworkExtrasBundle(AdMobAdapter.class, additionalRequestParams.toBundle())
+          .build();
+      InterstitialAd.load(getContext(), mAdUnitID, adRequest, new InterstitialAdLoadCallback() {
+        @Override
+        public void onAdLoaded(@NonNull InterstitialAd interstitialAd) {
+          mInterstitialAd = interstitialAd;
+          mInterstitialAd.setFullScreenContentCallback(new FullScreenContentCallback() {
+            @Override
+            public void onAdFailedToShowFullScreenContent(@NonNull AdError adError) {
+              super.onAdFailedToShowFullScreenContent(adError);
+            }
+
+            @Override
+            public void onAdShowedFullScreenContent() {
+              sendEvent(Events.DID_OPEN.toString(), new Bundle());
+              if (mShowAdPromise != null) {
+                mShowAdPromise.resolve(null);
+                mShowAdPromise = null;
+              }
+            }
+
+            @Override
+            public void onAdDismissedFullScreenContent() {
+              sendEvent(Events.DID_CLOSE.toString(), new Bundle());
+            }
+          });
+
+          sendEvent(Events.DID_LOAD.toString(), new Bundle());
+          if (mRequestAdPromise != null) {
+            mRequestAdPromise.resolve(null);
+            mRequestAdPromise = null;
           }
-          AdRequest adRequest = adRequestBuilder.build();
-          mInterstitialAd.loadAd(adRequest);
         }
-      }
+
+        @Override
+        public void onAdFailedToLoad(@NonNull LoadAdError loadAdError) {
+          sendEvent(
+              Events.DID_FAIL_TO_LOAD.toString(),
+              AdMobUtils.createEventForAdFailedToLoad(loadAdError));
+          if (mRequestAdPromise != null) {
+            mRequestAdPromise.reject(
+                "E_AD_REQUEST_FAILED",
+                AdMobUtils.errorStringForAdFailedCode(loadAdError.getCode()),
+                null);
+            mRequestAdPromise = null;
+          }
+        }
+      });
     });
   }
 
+
   @ExpoMethod
   public void showAd(final Promise promise) {
-    new Handler(Looper.getMainLooper()).post(new Runnable() {
-      @Override
-      public void run() {
-        if (mInterstitialAd != null && mInterstitialAd.isLoaded()) {
-          mShowAdPromise = promise;
-          mInterstitialAd.show();
-        } else {
-          promise.reject("E_AD_NOT_READY", "Ad is not ready", null);
-        }
+    new Handler(Looper.getMainLooper()).post(() -> {
+      if (mInterstitialAd != null) {
+        mShowAdPromise = promise;
+        mInterstitialAd.show(mActivityProvider.getCurrentActivity());
```

Comment:
Probably we should make sure that the current activity isn't `null`, the result is nullable

Reply:
Fair point, but as this is the last SDK we're supporting this module I would just leave it as is.
I guess there're lot more issues like this with module 👀